data-prepper-plugins/http-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/http/OAuthAccessTokenManager.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.sink.http;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.hc.client5.http.classic.methods.HttpPost;
+import org.apache.hc.client5.http.impl.classic.HttpClientBuilder;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.opensearch.dataprepper.plugins.sink.http.configuration.BearerTokenOptions;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Base64;
+import java.util.Map;
+
+public class OAuthAccessTokenManager {
+
+    public static final String BASIC = "Basic ";
+
+    public static final String BEARER = "Bearer ";
+
+    public static final String APPLICATION_X_WWW_FORM_URLENCODED = "application/x-www-form-urlencoded";
+    public static final String EXP = "exp";
+    public static final String ACCESS_TOKEN = "access_token";
+
+    public static final String REFRESH_TOKEN = "refresh_token";
+
+    private final ObjectMapper objectMapper;
+
+    private HttpClientBuilder httpClientBuilder;
+
+
+    public OAuthAccessTokenManager(final HttpClientBuilder httpClientBuilder){
+        this.httpClientBuilder = httpClientBuilder;
+        this.objectMapper = new ObjectMapper();
+    }
+
+    public String getAccessToken(final BearerTokenOptions bearerTokenOptions) {
+        HttpPost request = new HttpPost(bearerTokenOptions.getTokenURL());
+        request.setHeader(HttpHeaders.CONTENT_TYPE, APPLICATION_X_WWW_FORM_URLENCODED);
+        request.setHeader(HttpHeaders.AUTHORIZATION, BASIC + base64Encode(bearerTokenOptions.getClientId() + ":" + bearerTokenOptions.getClientSecret()));
+        String requestBody = "grant_type=" + bearerTokenOptions.getGrantType() +"&refresh_token"+bearerTokenOptions.getRefreshToken()+"&scope=" + bearerTokenOptions.getScope();
```

Comment:
Building strings like this can be difficult to avoid edge cases. Using an OAuth2 framework can help. Alternatively, there should be existing tools for creating POST form data correctly.

Reply:
Thanks for review @dlvenable , Sure I shall use library and avoid building strings this way